### PR TITLE
[RHPAM-3494] - Make pulling of dependencies and kjar validation skipp…

### PR DESCRIPTION
**Pull request by spolti**
…able in kie-server lauch script

Signed-off-by: spolti <fspolti@redhat.com>

### CODE REVIEW

File: pkg/controller/kieapp/defaults/defaults_test.go [hunk 1]
```diff
@@ -541,6 +541,84 @@ func TestRhdmProdImmutableEnvironment(t *testing.T) {
...
+					},
+				},
+			},
+		},
+	}
+	env, err := GetEnvironment(cr, test.MockService())
+	assert.Nil(t, err, "Error getting prod environment")
+
+	assert.Equal(t, "true", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_DISABLE_KC_VERIFICATION"), "Variable should not exist")
+	assert.Equal(t, "false", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_DISABLE_KC_PULL_DEPS"), "Variable should not exist")
```
Comment (jakubschwan) [inline pkg/controller/kieapp/defaults/defaults_test.go:567]:
```suggestion
	assert.Equal(t, "false", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_DISABLE_KC_PULL_DEPS"), "Variable should exist")
```

File: pkg/controller/kieapp/defaults/defaults_test.go [hunk 2]
```diff
@@ -541,6 +541,84 @@ func TestRhdmProdImmutableEnvironment(t *testing.T) {
...
+					},
+				},
+			},
+		},
+	}
+	env, err := GetEnvironment(cr, test.MockService())
+	assert.Nil(t, err, "Error getting prod environment")
+
+	assert.Equal(t, "true", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_DISABLE_KC_VERIFICATION"), "Variable should not exist")
+	assert.Equal(t, "true", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_DISABLE_KC_PULL_DEPS"), "Variable should not exist")
```
Comment (jakubschwan) [inline pkg/controller/kieapp/defaults/defaults_test.go:594]:
```suggestion
	assert.Equal(t, "true", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_DISABLE_KC_VERIFICATION"), "Variable should exist")
	assert.Equal(t, "true", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_DISABLE_KC_PULL_DEPS"), "Variable should exist")
```

**Review by jakubschwan — COMMENTED** [review]
In asserts should be updated messages for different cases, see suggestions

File: pkg/controller/kieapp/defaults/defaults_test.go (hunk 1 above)
Comment (spolti) [reply pkg/controller/kieapp/defaults/defaults_test.go:567]: Oh yes, thanks :)

File: pkg/controller/kieapp/defaults/defaults_test.go (hunk 2 above)
Comment (spolti) [reply pkg/controller/kieapp/defaults/defaults_test.go:594]: yeah, thanks :)

**Comment by tchughesiv** [comment]
@spolti what product version are you targeting?

**Comment by spolti** [comment]
@tchughesiv It is to be on 7.10.x and master for 7.11.
I was waiting review here first to cherry pick for 7.10.x branch.

**Comment by tchughesiv** [comment]
@spolti please reverse that plan. target 7.10.x first, then we can cherry pick to `main` for 7.11

**Review by jakubschwan — APPROVED** [review]
lgtm

**Comment by spolti** [comment]
@tchughesiv ok.

**Pull request merged**